dbt/adapters/databricks/constraints.py

```diff
@@ -0,0 +1,233 @@
+        warn_or_error(
+            ConstraintNotSupported(constraint=constraint.type.value, adapter="DatabricksAdapter")
+        )
+    elif constraint.warn_unenforced and not is_enforced(constraint):
+        warn_or_error(
+            ConstraintNotEnforced(constraint=constraint.type.value, adapter="DatabricksAdapter")
+        )
+
+    return supported
+
+
+def render_constraint(
+    constraint: T, render_funcs: dict[ConstraintType, RenderFunc]
+) -> Optional[str]:
+    rendered_constraint = ""
+
+    if constraint.type in render_funcs:
+        if constraint.name:
+            rendered_constraint = f"CONSTRAINT {constraint.name} "
+        rendered_constraint += render_funcs[constraint.type](constraint)
+
+    rendered_constraint = rendered_constraint.strip()
+
+    return rendered_constraint if rendered_constraint != "" else None
+
+
+def supported_for(constraint: T, support_func: SupportFunc, warning: str) -> bool:
+    if is_supported(constraint) and not support_func(constraint):
+        logger.warning(warning.format(type=constraint.type))
+
+    return is_supported(constraint) and support_func(constraint)
+
+
+# Shared render functions
+def render_error(constraint: ColumnLevelConstraint, missing: list[list[str]]) -> DbtValidationError:
+    fields = " or ".join(["(" + ", ".join([f"'{e}'" for e in x]) + ")" for x in missing])
+    constraint_type = constraint.type.value
+    constraint_name = "" if not constraint.name else f"{constraint.name} "
+    return DbtValidationError(
+        f"{constraint_type} constraint {constraint_name}is missing required field(s): {fields}"
+    )
+
+
+def render_custom(constraint: ColumnLevelConstraint) -> str:
+    assert constraint.type == ConstraintType.custom
+    if constraint.expression:
+        return constraint.expression
+    raise render_error(constraint, [["expression"]])
+
+
+# ColumnLevelConstraint specialization
+def parse_column_constraint(raw_constraint: dict[str, Any]) -> ColumnLevelConstraint:
+    return parse_constraint(ColumnLevelConstraint, raw_constraint)
+
+
+COLUMN_WARNING = (
+    "While constraint of type {type} is supported for models, it is not supported for columns."
+)
+
+supported_for_columns = partial(
+    supported_for, support_func=lambda x: x.type in SUPPORTED_FOR_COLUMN, warning=COLUMN_WARNING
+)
+
+
+def column_constraint_map() -> dict[ConstraintType, RenderFunc]:
+    return {
+        ConstraintType.primary_key: render_primary_key_for_column,
+        ConstraintType.foreign_key: render_foreign_key_for_column,
+        ConstraintType.custom: render_custom,
+    }
+
+
+def process_column_constraint(constraint: ColumnLevelConstraint) -> Optional[str]:
+    return process_constraint(constraint, supported_for_columns, column_constraint_map())
+
+
+def render_primary_key_for_column(constraint: ColumnLevelConstraint) -> str:
+    assert constraint.type == ConstraintType.primary_key
+    rendered = "PRIMARY KEY"
+    if constraint.expression:
```

Comment:
yes, because this is a column level constraint, so it is perfectly valid to just state that the column is a PRIMARY KEY.  In constrast, a model-level constraint needs to know which columns to treat as the primary key.  The support for expression is in case users want to add a constraint option such as DEFERRABLE.